Makefile

```diff
@@ -1,69 +1,70 @@
-SHELL = /usr/bin/env bash
-OS_NAME = $(shell uname -s)
-BUILD_CONFIGURATION = Debug
-BINARIES_PATH = $(shell pwd)/Binaries
-SRC_PATH = $(shell pwd)/src
-BOOTSTRAP_PATH = $(BINARIES_PATH)/Bootstrap
-BUILD_LOG_PATH =
-HOME_DIR = $(shell cd ~ && pwd)
-DOTNET_VERSION = 1.0.1
-DOTNET = $(BINARIES_PATH)/dotnet-cli/dotnet
-TARGET_FX = netcoreapp1.1
+SHELL := /usr/bin/env bash
+OS_NAME := $(shell uname -s)
+BUILD_CONFIGURATION := Debug
+THIS_MAKEFILE_PATH := $(dir $(realpath $(lastword $(MAKEFILE_LIST))))
+BINARIES_PATH := $(THIS_MAKEFILE_PATH)/Binaries
+SRC_PATH := $(THIS_MAKEFILE_PATH)/src
+BOOTSTRAP_PATH := $(BINARIES_PATH)/Bootstrap
+BUILD_LOG_PATH :=
+DOTNET_VERSION := 1.0.1
+DOTNET_PATH := $(BINARIES_PATH)/dotnet-cli
+DOTNET := $(DOTNET_PATH)/dotnet
+export PATH := $(DOTNET_PATH):$(PATH)
```

Comment:
Correct. However, we were setting `PATH` in every rule *anyway* (`export PATH="$(BINARIES_PATH)/dotnet-cli:$(PATH)"` at the start of every rule), so this has no effect. (Except, technically, the `$(DOTNET):` rule, but that *creates* `DOTNET_PATH`, so there's no harm on having it already there. Also `clean` I guess, but that's just an `rm`.)